ksql-engine/src/test/java/io/confluent/ksql/integration/StreamsSelectAndProjectAvroIntTest.java

```diff
@@ -0,0 +1,153 @@
+package io.confluent.ksql.integration;
+
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.test.IntegrationTest;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.KsqlContext;
+import io.confluent.ksql.serde.DataSource;
+import io.confluent.ksql.util.OrderDataProvider;
+import io.confluent.ksql.util.SchemaRegistryClientFactory;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+
+@Category({IntegrationTest.class})
+public class StreamsSelectAndProjectAvroIntTest {
+
+  private IntegrationTestHarness testHarness;
+  private KsqlContext ksqlContext;
+  private Map<String, RecordMetadata> recordMetadataMap;
+  private final String topicName = "TestTopic";
+  private OrderDataProvider dataProvider;
+
+  @Before
+  public void before() throws Exception {
+    testHarness = new IntegrationTestHarness(DataSource.DataSourceSerDe.AVRO.name());
```

Comment:
I wonder if it makes more sense to parameterize existing tests to use different Serdes? This seems very similar to existing tests but only with the Avro serde.

Reply:
That's a good point and I agree we should do it. We can refactor the tests in another PR.

Reply:
Perhaps its worth filing an issue for this and marking it as 'newbie'.